exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -270,89 +283,60 @@ def _translate_data(
-            max_value=self._MAX_RETRY_TIMEOUT
-        ):
-            if delay == self._MAX_RETRY_TIMEOUT or self._shutdown:
-                return self._result.FAILURE
-
-            with self._export_lock:
+        retry_info = RetryInfo()
+        with self._export_lock:
+            deadline_sec = time() + self._timeout
+            for retry_num in range(1, _MAX_RETRYS + 1):
+                backoff_seconds = 2 ** (retry_num - 1) * random.uniform(
+                    0.8, 1.2
+                )
                 try:
                     self._client.Export(
                         request=self._translate_data(data),
                         metadata=self._headers,
                         timeout=self._timeout,
                     )
-
                     return self._result.SUCCESS
-
                 except RpcError as error:
-                    if error.code() in [
-                        StatusCode.CANCELLED,
-                        StatusCode.DEADLINE_EXCEEDED,
-                        StatusCode.RESOURCE_EXHAUSTED,
-                        StatusCode.ABORTED,
-                        StatusCode.OUT_OF_RANGE,
-                        StatusCode.UNAVAILABLE,
-                        StatusCode.DATA_LOSS,
-                    ]:
-                        retry_info_bin = dict(error.trailing_metadata()).get(
-                            "google.rpc.retryinfo-bin"
-                        )
-                        if retry_info_bin is not None:
-                            retry_info = RetryInfo()
-                            retry_info.ParseFromString(retry_info_bin)
-                            delay = (
-                                retry_info.retry_delay.seconds
-                                + retry_info.retry_delay.nanos / 1.0e9
-                            )
-
-                        logger.warning(
-                            (
-                                "Transient error %s encountered while exporting "
-                                "%s to %s, retrying in %ss."
-                            ),
-                            error.code(),
-                            self._exporting,
-                            self._endpoint,
-                            delay,
+                    retry_info_bin = dict(error.trailing_metadata()).get(
+                        "google.rpc.retryinfo-bin"
+                    )
+                    if retry_info_bin is not None:
+                        retry_info.ParseFromString(retry_info_bin)
+                        backoff_seconds = (
+                            retry_info.retry_delay.seconds
+                            + retry_info.retry_delay.nanos / 1.0e9
                         )
-                        sleep(delay)
-                        continue
-                    else:
+                    if (
+                        error.code() not in _RETRYABLE_ERROR_CODES
+                        or retry_num == _MAX_RETRYS
```

Comment:
nit but imo it's easier to understand if the loop uses zero based indexing and then this is the only check that needs adjusting `retry_num == _MAX_RETRYS - 1`